dsl_parser/functions.py

```diff
@@ -234,20 +244,32 @@ def validate(self, plan):
                 "unknown input '{0}'.".format(input_value))
 
     def evaluate(self, handler):
+        return_value = None
         if isinstance(self.input_value, list):
             if any(is_function(x) for x in self.input_value):
                 raise exceptions.FunctionEvaluationError(
                     '{0}: found an unresolved argument: {1}'
                     .format(self.name, self.input_value))
 
-            return self._get_input_attribute(
+            return_value = self._get_input_attribute(
                 handler.get_input(self.input_value[0]))
 
         if is_function(self.input_value):
             raise exceptions.FunctionEvaluationError(
                 '{0}: found an unresolved argument: {1}'
                 .format(self.name, self.input_value))
-        return handler.get_input(self.input_value)
+        return_value = return_value or handler.get_input(self.input_value)
+
+        if self.return_type:
+            from dsl_parser.utils import parse_simple_type_value
```

Comment:
hmm unfortunate. We can untangle this by moving is_function and TEMPLATE_FUNCTIONS into utils.py...?

Reply:
not really, already tried that

Reply:
oh? I just moved them and it seems fine, `= 1051 passed, 1 warning in 9.26s =`

Reply:
You can try running it on the actual manager – I think that's where the problems began.  It was late on the other hand, I will check one more time